app/forms/serializers/submission.py

```diff
@@ -1,27 +1,55 @@
-from app.common.serializers import BaseModelSerializer
-from app.content.serializers import UserInAnswerSerializer
-from app.forms.models import Answer, Submission
-from app.forms.serializers import (
-    FieldInAnswerSerializer,
-    FormInSubmissionSerializer,
-    OptionSerializer,
-)
+from rest_framework import serializers
 
+from app.forms.models import Answer, Field, Option, Submission
+from app.forms.serializers import FieldInAnswerSerializer, OptionSerializer
 
-class AnswerSerializer(BaseModelSerializer):
-    field = FieldInAnswerSerializer(read_only=True)
-    selected_options = OptionSerializer(read_only=True)
+
+class AnswerSerializer(serializers.ModelSerializer):
+    field = FieldInAnswerSerializer()
+    selected_options = OptionSerializer(many=True, required=False)
 
     class Meta:
         model = Answer
         fields = ["id", "field", "selected_options", "answer_text"]
 
+    def validate(self, data):
+        if "selected_options" in data and "answer_text" in data:
+            raise serializers.ValidationError(
+                "Du kan ikke svare med både alternativer og tekst på samme spørsmål."
+            )
+        return data
+
 
-class SubmissionSerializer(BaseModelSerializer):
-    form = FormInSubmissionSerializer(read_only=True)
-    user = UserInAnswerSerializer(read_only=True)
-    answer = AnswerSerializer(read_only=True)
+class SubmissionSerializer(serializers.ModelSerializer):
+    answers = AnswerSerializer(many=True)
 
     class Meta:
         model = Submission
-        fields = ["form", "user", "answers"]
+        fields = ("answers",)
+
+    def create(self, validated_data):
```

Comment:
Denne kan med fordel brytes ned i flere metoder for at det skal bli mer clean

Reply:
Veldig godt poeng! :)